RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripRawToClusterAlgo.h

```diff
@@ -0,0 +1,221 @@
+            nStrip_wOfs += ((uint32_t)clustStripN) * num_bits / 8 + 2;
+            stripN += clustStripN;
+            // std::cout << (int)nStrip_Ofs << "," << (int)clustStripN << std::endl;
+          }
+        }
+        return stripN;
+      }
+
+      ALPAKA_FN_HOST_ACC inline uint8_t packetCode() const { return data_[(offset_ + 2) ^ 7]; }
+
+      ALPAKA_FN_HOST_ACC inline uint16_t cmMedian(uint8_t apvIndex) const {
+        uint16_t result = 0;
+        result |= data_[(offset_ + 3 + 2 * apvIndex) ^ 7];
+        result |= (((data_[(offset_ + 4 + 2 * apvIndex) ^ 7]) << 8) & 0x300);
+        return result;
+      }
+
+      ALPAKA_FN_HOST_ACC inline const uint8_t* data() const { return data_; }
+
+      ALPAKA_FN_HOST_ACC inline uint32_t offset() const { return offset_; }
+
+    private:
+      const uint8_t* data_;
+      uint32_t offset_;
+      uint16_t length_;
+      uint8_t headerLen_;
+    };
+  }  // namespace fedChannelDetails
+
+  class PortableFEDMover {
+  public:
+    PortableFEDMover(Queue* queue, uint32_t rawBufferSize, uint32_t fedChannelsNb)
+        : buffer(cms::alpakatools::make_host_buffer<uint8_t[]>(*queue, rawBufferSize)),
+          mapping(fedChannelsNb, *queue),
+          ofs(0),
+          chanN(0),
+          fedChannelsNb_(fedChannelsNb),
+          ofsFedId(sistrip::FED_ID_MAX + 1) {
+      // std::cout << "#portBuffSize," << rawBufferSize << std::endl;
+      queue_ = queue;
+    }
+
+    void fillBuffer(const std::vector<const FEDRawData*>& raw) {
+      for (uint16_t fedId = sistrip::FED_ID_MIN; fedId < (sistrip::FED_ID_MAX + 1); fedId++) {
+        if (raw[fedId]) {
+          std::memcpy(buffer.data() + ofs, raw[fedId]->data(), raw[fedId]->size());
+          ofsFedId[fedId] = ofs;
+          ofs += raw[fedId]->size();
+        }
+      }
+      // std::cout << "#fillBuffer," << ofs << std::endl;
+    }
+
+    void fillMapping(const std::vector<FEDChMetadata>& channelsMeta) {
+      for (uint32_t i = 0; i < fedChannelsNb_; i++, chanN++) {
+        const auto& channel = channelsMeta[i];
+        mapping->detID(i) = channel.detId;
+        mapping->fedID(i) = channel.fedId;
+        mapping->fedCh(i) = channel.fedCh;
+        mapping->fedChOfs(i) = channel.fedChOfs;
+        mapping->fedChDataOfsBuf(i) = channel.fedChOfs_wrt_rawFedId + ofsFedId[channel.fedId];
+        mapping->readoutMode(i) = channel.fedBufferRO;
+
+        // // Checks that I can rebuilt the FEDChannel
+        // bool isNonLite = (mapping->readoutMode(i) == 10 || mapping->readoutMode(i) == 11);
+        // auto ch = sistrip::fedChannelDetails::FEDChannel(buffer.data()+mapping->fedChDataOfsBuf(i), channel.fedChOfs, isNonLite);
+        // std::cout << "#makeMapping," << i << "," << (int)channel.fedId << "," << (int)channel.fedCh  << "," << ch.length() << "," << (int)channel.fedChOfs  << "," << channel.fedChOfs_wrt_rawFedId  << "," << (int)ch.stripsInCh<8>() << std::endl;
+      }
+    }
+
+    auto getBuffer() { return buffer; }
```

Comment:
while technically this could be marked `const`, returning a copy of the buffer allows modification of the data, so it should be left as non-`const`.